asserts/ifacedecls.go

```diff
@@ -235,10 +235,17 @@ func (c *AttributeConstraints) Check(attrs map[string]interface{}) error {
 	return c.matcher.match("", attrs)
 }
 
+// OnClassicConstraint specifies a constraint based whether the system is classic and optional specific distros' sets.
+type OnClassicConstraint struct {
+	Classic bool
+	Distros []string
+}
+
 // rules
 
 var (
 	validSnapType  = regexp.MustCompile("^(?:core|kernel|gadget|app)$")
+	validDistro    = regexp.MustCompile("^[-0-9a-z._]+$")
```

Comment:
This is new to the spec, but based on what I heard in the end of day summary, this seems ok. Seems slightly odd OTOH that we allow '-' and '_' but not 'A-Z', but I won't block on this.